### DISCUSSION TITLE
Unable to run published Browser WASM app, only runs using dotnet run

### DISCUSSION THREAD

**Discussion by Mrcubix** [AvaloniaUI/Avalonia#12862] [Q&A] [answered]
I've been trying publish and potentially host the Browser, with and without Blazor.

Both build successfully, however, whenever i try to run any web server in the generated AppBundle server and try to access the root URL, i'm greeted with the avalonia logo, and nothing else, except when opening the dev console, where i can see that this message got logged a couple times : 
```log
Your mono runtime and class libraries are out of sync.
dotnet.js:14 The out of sync library is: System.Private.CoreLib.dll
dotnet.js:14 
dotnet.js:14 When you update one from git you need to update, compile and install
dotnet.js:14 the other too.
dotnet.js:14 Do not report this as a bug unless you're sure you have updated correctly:
dotnet.js:14 you probably have a broken mono install.
dotnet.js:14 If you see other errors or faults after this message they are probably related
dotnet.js:14 and you need to fix your mono install first.
```

Followed by a couple of warning StackOverflowExceptions in `put_char` and then the following errors :

```log
Error: [MONO] * Assertion at /__w/1/s/src/mono/mono/mini/interp/interp.c:2119, condition `<disabled>' not met
    at xe (dotnet.js:5:509)
    at Object.je (dotnet.js:5:792)
    at _mono_wasm_trace_logger (dotnet.js:14:135708)
    at dotnet.wasm:0x344c2e
    at dotnet.wasm:0x40b5c7
    at dotnet.wasm:0x1f1696
    at dotnet.wasm:0x316b6
    at dotnet.wasm:0x13b21
    at dotnet.wasm:0x91e2
    at dotnet.wasm:0x3d2c60
```
    
Anything i can do to even run the app? so i can potentially fix an issue with invariant cultures exceptions only happening in release (it worked once then never did again)

(Also the app doesn't even try to run at all on github since for some reasons, it throws 404 errors for any other files with a dot in their path, removed them, will see if it's any different)

**Comment by timunie** [comment]
could it be that the server is missing the correct .NET version installed? Does it help if you choose a different version of .NET to publish the project? 

https://gprivate.com/66mi1

**Reply by Mrcubix** [reply]
Tried with : 
- .NET 7.0.400
- .NET 7.0.101 (as specified in global.json)

Haven't tried with .NET 6 yet if that's even possible anymore

**Comment by maxkatz6** [comment, marked as answer]
How exactly are you running this app? Does it work with dotnet serve?
About github pages, cc @wieslawsoltes for help.

**Reply by Mrcubix** [reply]
Starting most web servers one folder before the AppBundle, then navigating to that folder in a browser (firefox & chrome).
~~Haven't tried with "dotnet serve", whatever that is (nothing about it online at first glance), gonna look around more and let you know~~. It works with dotnet serve, but that won't do for github pages.
One thing i have noticed is that no AppBundle folder is copied to the publish folder. No index file are also present within.

Update: for the github page, Managed to get the default avalonia splashscreen to show up but now need to fix this exception somehow:

```log
Uncaught ManagedError: AggregateException_ctor_DefaultMessage (Argument_CultureNotSupportedInInvariantMode Arg_ParamName_Name, name
Argument_CultureInvalidIdentifier, en-US)
```

Turns out localization might not work as i would want it to on browser.

Update II: Managed to fix that one, now getting this exception:

![image](https://github.com/AvaloniaUI/Avalonia/assets/39861216/98d837f9-a259-41b5-8daf-75268af992c1)

This issue doesn't happen on desktop windows, android & running the browser app on my PC. The path match exactly with my file structure.

![image](https://github.com/AvaloniaUI/Avalonia/assets/39861216/5f9208fc-9b5f-4076-99e5-24ce248d5a0a)

The app work in WSL but getting the same issue on my pi.

For some reasons, Avalonia want the the `Tilesets` word to be lowercase on desktop linux, even though it doesn't match the actual file path, else it won't work, seems like a platform specific bug to me.

I can't set a fallback path properly either since a WASM app, even if running on linux, isn't considered as linux for Avalonia, as reflected by `OperatingSystem.IsLinux()` returns false.

Easiest solution would be to catch the exception and try with the lowercase word, but that would be a dirty way to do it.

**Reply by timunie** [reply]
I told you before to mind asset names need to be case senstitive. You doubt it, and here we are. Check in your sample that `tileset` folder name is actually lowercase. 
![image](https://github.com/AvaloniaUI/Avalonia/assets/47110241/1b429728-83d4-4337-949d-e32757182cb0)

**Reply by Mrcubix** [reply]
Check the screenshot above where it is capsulated

I guess git committed that one folder with an lowercase t, not unusual

After cleaning that up by removed the cache which removed both and re-adding it again, it work
